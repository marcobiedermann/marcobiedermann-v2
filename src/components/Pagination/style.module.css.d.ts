/* eslint-disable @typescript-eslint/camelcase */

export const pagination: string;
export const pagination__link: string;
