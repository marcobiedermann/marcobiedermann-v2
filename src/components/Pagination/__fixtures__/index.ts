const next = {
  id: 'next',
  name: 'Next',
  path: '/next',
};

const previous = {
  id: 'previous',
  name: 'Previous',
  path: '/previous',
};

export { next, previous };
