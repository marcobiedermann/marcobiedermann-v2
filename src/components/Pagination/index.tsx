import classNames from 'classnames';
import { Link } from 'gatsby';
import React from 'react';
import { Route } from '../../constants/routes';
import styles from './style.module.css';

export interface PaginationProps {
  className?: string;
  routes: Route[];
}

const Pagination: React.FC<PaginationProps> = props => {
  const { className, routes, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.pagination)} {...otherProps}>
      {routes.map(route => (
        <li key={route.path}>
          <Link to={route.path} className={styles.pagination__link}>
            {route.name}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default Pagination;
