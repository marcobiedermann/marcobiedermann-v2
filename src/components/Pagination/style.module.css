.pagination {
  display: grid;
  gap: 0.5em;
  grid-auto-flow: column;
  justify-content: center;
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 1.5em;
  padding: 0;
}

.pagination__link {
  border: 1px solid;
  display: inline-block;
  min-width: 2.33em;
  padding: 0.25em 0.5em;
  text-align: center;
}
