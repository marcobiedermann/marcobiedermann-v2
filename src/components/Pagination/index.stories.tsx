/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Pagination from '.';
import { next, previous } from './__fixtures__';

storiesOf('Pagination', module).add('default', () => <Pagination next={next} previous={previous} routes={[]} />);
