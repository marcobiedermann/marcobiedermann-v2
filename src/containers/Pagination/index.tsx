import { navigate } from '@reach/router';
import React from 'react';
import { GlobalHotKeys } from 'react-hotkeys';
import Pagination, { PaginationProps } from '../../components/Pagination';

const keyMap = {
  NAVIGATE_PREVIOUS: 'left',
  NAVIGATE_NEXT: 'right',
};

const PaginationContainer: React.FC<PaginationProps> = props => {
  const handlers = {
    NAVIGATE_PREVIOUS: (): void => {
      const { previous } = props;

      return previous && navigate(previous.path);
    },
    NAVIGATE_NEXT: (): void => {
      const { next } = props;

      return next && navigate(next.path);
    },
  };

  return (
    <GlobalHotKeys handlers={handlers} keyMap={keyMap}>
      <Pagination {...props} />
    </GlobalHotKeys>
  );
};

export default PaginationContainer;
