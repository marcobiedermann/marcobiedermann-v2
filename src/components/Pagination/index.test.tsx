/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '.';

describe('Pagination component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Pagination routes={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render routes correctly', () => {
    const routes = [
      {
        name: 'Previous',
        path: '/previous',
      },
      {
        name: 'Next',
        path: '/next',
      },
    ];
    const tree = renderer.create(<Pagination routes={routes} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
