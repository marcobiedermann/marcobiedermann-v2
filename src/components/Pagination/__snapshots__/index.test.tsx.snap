// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render correctly 1`] = `
<ul
  className="pagination"
/>
`;

exports[`Pagination component should render routes correctly 1`] = `
<ul
  className="pagination"
>
  <li>
    <a
      className="pagination__link"
      href="/previous"
    >
      Previous
    </a>
  </li>
  <li>
    <a
      className="pagination__link"
      href="/next"
    >
      Next
    </a>
  </li>
</ul>
`;
